docs/source/developer/helpers/result.rst

```diff
@@ -0,0 +1,256 @@
+generated dicts whenever you need them as results.
+
+..  code-block:: python
+
+    from fotoobo.helpers import result
+    res = result.Result()
+    res.push_result("host1", {
+        "version": 1,
+        "build": 2000,
+        "host": "host1.local"
+    })
+    res.push_result("host2", {
+        "version": 2,
+        "build": 2020,
+        "host": "host2.local"
+    })
+    res.all_results()
+
+..  code-block:: text
+
+    {'host1': {'version': 1, 'build': 2000, 'host': 'host1.local'}, 'host2': {'version': 2, 'build': 2020, 'host': 'host2.local'}}
+
+..  code-block:: python
+
+    res.print_result_as_table()
+
+..  code-block:: text
+
+    ┌───────┬───┬──────┬─────────────┐
+    │ host1 │ 1 │ 2000 │ host1.local │
+    │ host2 │ 2 │ 2020 │ host2.local │
+    └───────┴───┴──────┴─────────────┘
+
+You see that the keys of the dicts are not visible with standard output. Use the
+:code:`auto_header` argument the show the dict keys as column headers.
+
+..  code-block:: python
+
+    res.print_result_as_table(auto_header=True)
+
+..  code-block:: text
+
+    ┏━━━━━━━┳━━━━━━━━━┳━━━━━━━┳━━━━━━━━━━━━━┓
+    ┃ key   ┃ version ┃ build ┃ host        ┃
+    ┡━━━━━━━╇━━━━━━━━━╇━━━━━━━╇━━━━━━━━━━━━━┩
+    │ host1 │ 1       │ 2000  │ host1.local │
+    │ host2 │ 2       │ 2020  │ host2.local │
+    └───────┴─────────┴───────┴─────────────┘
+
+Of course you may also use the :code:`headers` argument if you'd like to customize header names.
+
+Results with dict when dicts are not consistent
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+With the magic we use with dicts in results you'll get unwanted output if you mix types or change
+the key names or order.
+
+..  code-block:: python
+
+    from fotoobo.helpers import result
+    res = result.Result()
+    res.push_result("host1", {
+        "version": 1,
+        "build": 2000,
+        "host": "host1.local"
+    })
+    res.push_result("host2", {
+        "host": "host2.local",
+        "version": 2
+    })
+    res.push_result("host3", "text")
+    res.all_results()
+
+..  code-block:: text
+
+    {'host1': {'version': 1, 'build': 2000, 'host': 'host1.local'}, 'host2': {'host': 'host2.local', 'version': 2}, 'host3': 'text'}
+
+..  code-block:: python
+
+    res.print_result_as_table(auto_header=True)
+
+..  code-block:: text
+
+    ┏━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━┳━━━━━━━━━━━━━┓
+    ┃ key   ┃ version     ┃ build ┃ host        ┃
+    ┡━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━╇━━━━━━━━━━━━━┩
+    │ host1 │ 1           │ 2000  │ host1.local │
+    │ host2 │ host2.local │ 2     │             │
+    │ host3 │ text        │       │             │
+    └───────┴─────────────┴───────┴─────────────┘
+
+You would expect a table with looks like the line for host1. But for host2 you recognize that the
+hostname is in the wrong column. This is because the order of the key value pairs int the dict does
+not match the ordering of the first result. With host3 which not even has a dict als value you see
```

Comment:
... not even has a dict **as** value ...

Reply:
fixed